hashintel/hash
# Expose a Graph run configuration that only serves the OpenAPI Spec

## 🌟 What is the purpose of this PR?

We generate an OpenAPI client for the Graph. To do this, we currently have to start up all of the dependencies of the graph as it tries to connect to the database when it starts.

This PR adds a new argument `--openapi-only` which allows the Graph to start without the DB running, it only serves the OpenAPI route.

This allows `generate-openapi-client` to run much faster now, and reduces the risk of affecting local DBs in development when running it.

## ❓ How to test this?

- Check CI passes
- Try running the graph with `just run server --openapi-only` and make sure it starts, runs, serves the OpenAPI spec on `http://localhost:4000/api-doc/openapi.json`
- Make sure `just generate-openapi-client` works as expected
